Integrations/Elasticsearch_v2_5_5/Elasticsearch_v2_5_5.py

```diff
@@ -0,0 +1,640 @@
+            hit_timestamp = int(hit.get('_source')[str(TIME_FIELD)])
+
+            if hit_timestamp > last_fetch:
+                last_fetch = hit_timestamp
+
+            # avoid duplication due to weak time query
+            if hit_timestamp > current_fetch:
+                inc = {
+                    'name': 'Elasticsearch: Index: ' + str(hit.get('_index')) + ", ID: " + str(hit.get('_id')),
+                    'rawJSON': json.dumps(hit),
+                    'labels': incident_label_maker(hit.get('_source')),
+                    'occurred': hit_date.isoformat() + 'Z'
+                }
+                incidents.append(inc)
+
+    return incidents, last_fetch
+
+
+def results_to_incidents_datetime(response, last_fetch):
+    """Converts the current results into incidents.
+
+    Args:
+        response(dict): the raw search results from Elasticsearch.
+        last_fetch(datetime): the date or timestamp of the last fetch before this fetch
+        - this will hold the last date of the incident brought by this fetch.
+
+    Returns:
+        (list).The incidents.
+        (datetime).The date of the last incident brought by this fetch.
+    """
+    current_fetch = last_fetch
+    incidents = []
+    for hit in response.get('hits', {}).get('hits'):
+        if hit.get('_source') is not None and hit.get('_source').get(str(TIME_FIELD)) is not None:
+            hit_date = datetime.strptime(str(hit.get('_source')[str(TIME_FIELD)]), TIME_FORMAT)
+
+            if hit_date > last_fetch:
+                last_fetch = hit_date
+
+            # avoid duplication due to weak time query
+            if hit_date > current_fetch:
+                inc = {
+                    'name': 'Elasticsearch: Index: ' + str(hit.get('_index')) + ", ID: " + str(hit.get('_id')),
+                    'rawJSON': json.dumps(hit),
+                    'labels': incident_label_maker(hit.get('_source')),
+                    'occurred': hit_date.isoformat() + 'Z'
+                }
+                incidents.append(inc)
+
+    return incidents, last_fetch
+
+
+def get_indicators_command():
+    search, _ = get_indicators_search_scan()
+    limit = int(demisto.args().get('limit', FETCH_SIZE))
+    indicators_list: list = []
+    for hit in search.scan():
+        indicators_list.extend(extract_indicators_from_insight_hit(hit))
+        if len(indicators_list) >= limit:
+            break
+    hr = tableToMarkdown('Indicators', indicators_list, ['name'])
+    return_outputs(hr, {'ElasticsearchFeed.SharedIndicators': indicators_list}, indicators_list)
+
+
+def fetch_indicators_command():
+    search, now = get_indicators_search_scan()
+    ioc_lst: list = []
+    for hit in search.scan():
+        ioc_lst.extend(extract_indicators_from_insight_hit(hit))
+    if ioc_lst:
+        for b in batch(ioc_lst, batch_size=2000):
+            demisto.createIndicators(b)
+    demisto.setLastRun({'time': now})
```

Comment:
are we protecting against running fetch indicators and fetch incidents from the same instance?
because if not, we can't just use `demisto.setLastRun({'time': now})` for both.

Reply:
It's impossible to have the same instance fetch-incidents and fetch-indicators, the user has to choose which one to use.